contracts/upgradeable_contracts/omnibridge_nft/BasicNFTOmnibridge.sol

```diff
@@ -157,77 +173,128 @@ abstract contract BasicNFTOmnibridge is
      */
-    function fixMediatorBalance(
+    function fixMediatorBalanceERC721(
         address _token,
         address _receiver,
-        uint256 _tokenId
+        uint256[] calldata _tokenIds
     ) external onlyIfUpgradeabilityOwner {
         require(isRegisteredAsNativeToken(_token));
-        require(!mediatorOwns(_token, _tokenId));
-        require(IERC721(_token).ownerOf(_tokenId) == address(this));
+        require(_tokenIds.length > 0);
 
-        _setMediatorOwns(_token, _tokenId, true);
+        uint256[] memory values = new uint256[](0);
 
-        bytes memory data = _prepareMessage(_token, _receiver, _tokenId);
+        bytes memory data = _prepareMessage(_token, _receiver, _tokenIds, values);
         bytes32 _messageId = _passMessage(data, true);
-        _recordBridgeOperation(_messageId, _token, _receiver, _tokenId);
+        _recordBridgeOperation(_messageId, _token, _receiver, _tokenIds, values);
+    }
+
+    /**
+     * @dev Allows to send to the other network some ERC1155 token that can be forced into the contract
+     * without the invocation of the required methods.
+     * Before calling this method, it must be carefully investigated how imbalance happened
+     * in order to avoid an attempt to steal the funds from a token with double addresses.
+     * @param _token address of the token contract.
+     * @param _receiver the address that will receive the token on the other network.
+     * @param _tokenIds unique ids of the bridged tokens.
+     * @param _values corresponding amounts of the bridged tokens.
+     */
+    function fixMediatorBalanceERC1155(
+        address _token,
+        address _receiver,
+        uint256[] calldata _tokenIds,
+        uint256[] calldata _values
+    ) external onlyIfUpgradeabilityOwner {
+        require(isRegisteredAsNativeToken(_token));
+        require(_tokenIds.length == _values.length);
+        require(_tokenIds.length > 0);
+
+        bytes memory data = _prepareMessage(_token, _receiver, _tokenIds, _values);
+        bytes32 _messageId = _passMessage(data, true);
+        _recordBridgeOperation(_messageId, _token, _receiver, _tokenIds, _values);
     }
 
     /**
      * @dev Executes action on deposit of ERC721 token.
      * @param _token address of the ERC721 token contract.
      * @param _from address of token sender.
      * @param _receiver address of token receiver on the other side.
-     * @param _tokenId unique id of the bridged token.
+     * @param _tokenIds unique ids of the bridged tokens.
+     * @param _values amounts of bridged tokens. Should be empty list for ERC721.
      */
     function bridgeSpecificActionsOnTokenTransfer(
         address _token,
         address _from,
         address _receiver,
-        uint256 _tokenId
+        uint256[] memory _tokenIds,
+        uint256[] memory _values
     ) internal override {
-        // verify that token was indeed transferred
-        require(IERC721(_token).ownerOf(_tokenId) == address(this));
```

Comment:
Was it moved to another function or was removed by mistake?

Reply:
Was moved to `_prepareMessage`.